lib/src/kvs/tests/multiwriter.rs

```diff
@@ -7,14 +7,17 @@ async fn multiwriter_same_key() {
 	let mut tx = ds.transaction(true, false).await.unwrap();
 	tx.set("test", "some text").await.unwrap();
 	tx.commit().await.unwrap();
-	// Create a writeable transaction
+    // Create a writeable transaction
 	let mut tx1 = ds.transaction(true, false).await.unwrap();
+	tx1.get("test").await.unwrap(); // adds a read dependency to "test"
 	tx1.set("test", "other text").await.unwrap();
 	// Create a writeable transaction
 	let mut tx2 = ds.transaction(true, false).await.unwrap();
+	tx2.get("test").await.unwrap(); // adds a read dependency to "test"
 	tx2.set("test", "other text").await.unwrap();
 	// Create a writeable transaction
 	let mut tx3 = ds.transaction(true, false).await.unwrap();
+	tx3.get("test").await.unwrap(); // adds a read dependency to "test"
```

Comment:
I'm worried that in order to make this test work, we need to add read dependencies before writing a key. This might not happen outside of a test, and then the expected behaviour doesn't occur. Do you know why this is the case?

Reply:
@tobiemh There is an internal discussion on this. @mumoshu is in a better position to answer this but if you would like to keep the previous behaviour for other engines, I can change `tx2.commit().await.is_err()` to `tx2.commit().await.is_ok()` just for FoundationDB only since it should never error here by design.

Reply:
@tobiemh @rushmorem As I've noted internally, I presume this is the expected behavior for FDB!
I also believe this is the "desired" behavior for other DBs but unfortunately, not due to how their transactions work.

That said, if we are to accept and ensure that there's this difference between FDB and others, I'd suggest having two separate test files, `multiwriter_same_key_fdb.rs` and `multiwriter_same_key_default.rs` each has its own `multiwriter_same_key` function. All other test functions in the current `multiwriter.rs` should remain within `multiwriter.rs` .

That way, we could state in the code that there's a difference and that's tested/expected.
WDYT?

Reply:
@mumoshu If the difference between those files is going to end up just being those reads, we can feature-gate just those three lines for FoundationDB.